### PR TITLE
chore(staff): Create sentry apps/staff perm class and attach to base endpoint

### PR DESCRIPTION
**Pull request by schew2381**
There are only 2 endpoints that use `SentryAppsBaseEndpoint` which are `SentryAppsEndpoint` and `SentryAppsStatsEndpoint`.

`SentryAppsEndpoint` allows you to fetch unpublished/published integrations and also create new ones.
We only allow staff to use the GET and disable the POST through a new class variable on the `StaffPermissionMixin`. 
```Python
class StaffPermissionMixin:
    staff_allowed_methods  = {"GET", "PUT", "POST", "DELETE"}

    def has_permission(...):
        if request.method in staff_allowed_methods and is_active_staff(request):
            return True
        return super().has_permission(...)

    def has_obj_permission(...):
        # same thing basically
        if request.method in staff_allowed_methods ....

class SentryAppsAndStaffPermission(StaffPermissionMixin, SentryAppsPermission):
    staff_allowed_methods = {"GET"}
```
Honestly, I should've done this sooner and will probably have to go back to some staff endpoint request methods that should not be allowed by staff.

### CODE REVIEW

File: src/sentry/api/endpoints/system_options.py [hunk 1]
```diff
@@ -80,10 +80,7 @@ def has_permission(self, request: Request):
             return True
         if not request.access.has_permission("options.admin"):
             # We ignore options.admin permission is all keys in the update match the allowlist.
-            if all([k in SYSTEM_OPTIONS_ALLOWLIST for k in request.data.keys()]):
-                return True
-
-            return False
+            return all([k in SYSTEM_OPTIONS_ALLOWLIST for k in request.data.keys()])
```
Comment (schew2381) [inline src/sentry/api/endpoints/system_options.py:83]: This is a random thing I saw and wanted to fix :O

File: src/sentry/api/endpoints/integrations/sentry_apps/index.py [hunk 2]
```diff
@@ -88,9 +90,9 @@ def post(self, request: Request, organization) -> Response:
             "schema": request.json_body.get("schema", {}),
             "overview": request.json_body.get("overview"),
             "allowedOrigins": request.json_body.get("allowedOrigins", []),
-            "popularity": request.json_body.get("popularity")
-            if is_active_superuser(request)
-            else None,
+            "popularity": (
+                request.json_body.get("popularity") if is_active_superuser(request) else None
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/integrations/sentry_apps/index.py:94]: should this use `elevated_user` as well?

Comment (schew2381) [reply src/sentry/api/endpoints/integrations/sentry_apps/index.py:94]: Staff won't have access to the POST so it should be fine to leave as just superuser

**Review by JoshFerge — COMMENTED** [review]
could we make sure we have 100% test coverage on this once since it pertains to security? see a few lines w/ missing coverage

File: src/sentry/testutils/cases.py [hunk 3]
```diff
@@ -644,16 +644,21 @@ def detect_performance_problems_interceptor(data: Event, project: Project):
...
+            mock.patch(
+                "sentry.event_manager.detect_performance_problems",
+                side_effect=detect_performance_problems_interceptor,
+            ),
+            mock.patch.object(
+                issue_type, "noise_config", new=NoiseConfig(noise_limit, timedelta(minutes=1))
+            ),
+            override_options(
+                {"performance.issues.all.problem-detection": 1.0, detector_option: 1.0}
+            ),
```
Comment (cathteng) [inline src/sentry/testutils/cases.py:661]: 😵‍💫

File: tests/sentry/api/endpoints/test_sentry_apps.py [hunk 4]
```diff
@@ -162,39 +157,55 @@ def make_token_request(
...
+        assert self.published_app.uuid in response_uuids
+        assert self.unpublished_app.uuid in response_uuids
+        assert self.unowned_unpublished_app.uuid in response_uuids
+
+        with self.settings(SENTRY_SELF_HOSTED=False):
+            self.get_success_response(status_code=200)
 
     def test_superuser_sees_all_apps(self):
-        response = self.get_success_response()
+        response = self.get_success_response(status_code=200)
```
Comment (cathteng) [inline tests/sentry/api/endpoints/test_sentry_apps.py:183]: tip: 200 is implied if you don't pass it in, but it's fine to change

File: tests/sentry/api/endpoints/test_sentry_apps.py [hunk 5]
```diff
@@ -343,15 +365,15 @@ def test_superuser_read_cannot_create(self):
     @with_feature("auth:enterprise-superuser-read-write")
     def test_superuser_read_can_create(self):
         self.add_user_permission(self.superuser, "superuser.write")
-        self.get_success_response(**self.get_data(popularity=POPULARITY))
+        self.get_success_response(**self.get_data(popularity=POPULARITY), status_code=201)
```
Comment (cathteng) [inline tests/sentry/api/endpoints/test_sentry_apps.py:368]: tip: 201 is implied for POST unless you specific otherwise

**Review by cathteng — APPROVED** [review]
the method check is a good idea 👍

**Pull request merged**